src/store/slicers/filtersSlice.ts

```diff
@@ -0,0 +1,66 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { ENDPOINTS } from "../../utils/types";
+
+const initialState = {
+  endpoint: "top-headlines",
+  searchInput: "",
+  source: "",
+  dateFrom: "",
+  dateTo: "",
+  language: "",
+  sortBy: "",
+  country: "",
+  category: "",
+};
+const filtersSlice = createSlice({
```

Comment:
Great implementation.
You may union some setFunc to just updateFIlter object.
Do not change, just think about it.